types/react/index.d.ts

```diff
@@ -354,7 +354,7 @@ declare namespace React {
         displayName?: string;
     }
 
-    interface ComponentClass<P = {}> extends StaticLifecycle<P, any> {
+    interface ComponentClass<P = {}, S = any> extends StaticLifecycle<P, S> {
```

Comment:
Curious, if you try using `S = {}` as the generic param, does it break any tests? I think `S = {}` is probably better since `Component.prototype.setState` does shallow renders against a JS object.

Reply:
let's see